src/org/thoughtcrime/securesms/PinnedMessageAdapter.java

```diff
@@ -84,17 +90,37 @@ public ViewHolder(View v) {
         return new ViewHolder(theInflatedView);
     }
 
-
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
         dataCursor.moveToPosition(position);
-        MmsSmsDatabase.Reader reader = db.readerFor(dataCursor, masterSecret);
-        MessageRecord         record = reader.getCurrent();
+        ConversationItem      conversationItem = new ConversationItem(context);
+        MmsSmsDatabase.Reader reader           = db.readerFor(dataCursor, masterSecret);
+        MessageRecord         record           = reader.getCurrent();
+
         this.setMessageView(record, holder);
 
-        if (record.isMms() && ((MmsMessageRecord) record).getSlideDeck().getSlides().size() != 0) {
-            ConversationItem                        conversationItem       = new ConversationItem(context);
-            ConversationItem.ThumbnailClickListener thumbnailClickListener = conversationItem.new ThumbnailClickListener(record);
+        if (isDocument(record)) {
+            ConversationItem.ThumbnailClickListener thumbnailClickListener
+                    = conversationItem.new ThumbnailClickListener(record);
+            ConversationItem.AttachmentDownloadClickListener attachmentDownloadClickListener
+                    = conversationItem.new AttachmentDownloadClickListener();
+            holder.documentViewStub.get().setDocument(((MediaMmsMessageRecord)record).getSlideDeck().getDocumentSlide(), true);
+            holder.documentViewStub.get().setDocumentClickListener(thumbnailClickListener);
+            holder.documentViewStub.get().setDownloadClickListener(attachmentDownloadClickListener);
+            holder.documentViewStub.get().setVisibility(view.VISIBLE);
+        }
+
+        if (isAudio(record)) {
+            ConversationItem.AttachmentDownloadClickListener attachmentDownloadClickListener
+                    = conversationItem.new AttachmentDownloadClickListener();
+            holder.audioViewStub.get().setAudio(masterSecret, ((MediaMmsMessageRecord)record).getSlideDeck().getAudioSlide(), true);
+            holder.audioViewStub.get().setDownloadClickListener(attachmentDownloadClickListener);
+            holder.audioViewStub.get().setVisibility(view.VISIBLE);
+        }
+
+        else if (isVideo(record)) {
```

Comment:
IMO this should be on the same line as the closing bracket.

https://github.com/signalapp/Signal-Android/search?utf8=%E2%9C%93&q=else+if&type=Code

Reply:
interesting
same goes for try catch